pkg/master/controller.go

```diff
@@ -60,6 +60,7 @@ type Controller struct {
 	ServiceIP         net.IP
 	ServicePort       int
 	PublicServicePort int
+	ServiceEnableDNS  bool
 
```

Comment:
It would be clearer to accept a list of additional service ports here instead of tying it to DNS specifically.  Actually, just passing all master ports in this way would be a better factoring.